ui/src/components/dataset/AssociatedProjects.vue

```diff
@@ -0,0 +1,293 @@
+      id: obj?.user?.id,
+      username: obj?.user?.username,
+    }));
+    // const contact_values = (contacts || []).map(
+    //   (contact) => contact?.contact?.value
+    // );
+    const _datasets = (datasets || []).map((d) => ({
+      id: d?.dataset?.id,
+      name: d?.dataset?.name,
+    }));
+    return {
+      ...rest,
+      users: _users,
+      // contacts: contact_values,
+      datasets: _datasets,
+    };
+  });
+});
+
+// This could've been split into two variables user_columns and admin_columns
+// and concataned based on user roles
+// but the order of columns would not be preserved
+
+const columns = [
+  { key: "name", sortable: true },
+  { key: "users", sortable: false, width: "30%" },
+  {
+    key: "datasets",
+    sortable: false,
+    width: "80px",
+    thAlign: "center",
+    tdAlign: "center",
+  },
+  { key: "created_at", label: "Created At", sortable: true, width: "100px" },
+  { key: "updated_at", label: "Updated At", sortable: true, width: "150px" },
+  { key: "assigned_at", label: "Assigned At", sortable: true, width: "150px" },
+  { key: "assigner", label: "Assigner", sortable: false, width: "150px" },
+];
+
+function fetch_projects() {
+  data_loading.value = true;
+
+  const skip = (params.value.currentPage - 1) * params.value.itemsPerPage;
+
+  const queryparams = {
+    search: params.value.search,
+    take: params.value.itemsPerPage,
+    skip: skip,
+    sortBy: params.value.sortBy,
+    sort_order: params.value.sortingOrder,
+  };
+
+  const IntdatasetID = parseInt(props.datasetId, 10);
+
+  DatasetService.getProjects({ id: IntdatasetID, params: queryparams })
+    .then((res) => {
+      console.log(res.data.projects);
+      projects.value = res.data.projects.map((project) => {
+        const matchingDataset = project.datasets.find(
+          (d) => d.dataset.id === IntdatasetID,
+        );
+        const assigned_at = matchingDataset
+          ? matchingDataset.assigned_at
+          : null;
+        const assigner =
+          matchingDataset && matchingDataset.assignor
+            ? matchingDataset.assignor.name
+            : "No Data";
+
+        return {
+          ...project,
+          assigned_at,
+          assigner,
+        };
+      });
+      totalItems.value = res.data.metadata.count;
+    })
+    .catch((error) => {
+      console.error(
+        "Error fetching projects:",
+        error.response || error.message,
+      );
+      toast.error("Error fetching projects");
+    })
+    .finally(() => {
+      data_loading.value = false;
+    });
+}
+
+fetch_projects();
+
+watch(
+  () => params.value.currentPage,
+  () => {
+    fetch_projects();
+  },
+);
+
+watch(
+  [
+    () => params.value.itemsPerPage,
+    () => params.value.search,
+    () => params.value.sortBy,
+    () => params.value.sortingOrder,
+  ],
+  () => {
+    if (params.value.currentPage === 1) {
+      fetch_projects();
+    }
+    params.value.currentPage = 1;
+  },
+);
```

Comment:
Add a watcher here which watches for changes in `params`. Emit an event to make the parent component aware of the fact that the search parameters have changes.

Then, in your parent component, catch this event, and use the updated params that the above event emitted to re-fetch the projects. This can be accomplished simply by moving your these 2 existing watchers to Dataset.vue